src/mesido/workflows/emerge.py

```diff
@@ -0,0 +1,246 @@
+import time
+
+import casadi as ca
+
+from mesido.esdl.esdl_additional_vars_mixin import ESDLAdditionalVarsMixin
+from mesido.esdl.esdl_mixin import ESDLMixin
+from mesido.esdl.esdl_parser import ESDLFileParser
+from mesido.esdl.profile_parser import ProfileReaderFromFile
+from mesido.head_loss_class import HeadLossOption
+from mesido.techno_economic_mixin import TechnoEconomicMixin
+
+from rtctools.optimization.collocated_integrated_optimization_problem import (
+    CollocatedIntegratedOptimizationProblem,
+)
+from rtctools.optimization.goal_programming_mixin_base import Goal
+from rtctools.optimization.linearized_order_goal_programming_mixin import (
+    LinearizedOrderGoalProgrammingMixin,
+)
+from rtctools.optimization.single_pass_goal_programming_mixin import SinglePassGoalProgrammingMixin
+from rtctools.util import run_optimization_problem
+
+
+class MaxRevenue(Goal):
+
+    priority = 1
+
+    order = 1
+
+    def __init__(self, asset_name: str):
+        """
+        The constructor of the goal.
+
+        Parameters
+        ----------
+        source : string of the source name that is going to be minimized
+        """
+        # self.target_max = 0
+        # self.function_range = (-1.0e9, 1.0e9)
+        # self.function_nominal = 1.0e7
+
+        self.asset_name = asset_name
+
+    def function(
+        self, optimization_problem: CollocatedIntegratedOptimizationProblem, ensemble_member: int
+    ) -> ca.MX:
+        """
+        This function returns the state variable to be minimized.
+
+        Parameters
+        ----------
+        optimization_problem : The optimization class containing the variables'.
+        ensemble_member : the ensemble member.
+
+        Returns
+        -------
+        The negative hydrogen production state of the optimization problem.
+        """
+        # TODO: not yet scaled
```

Comment:
Lets ensure it is properly scaled before it is merged to main.

Reply:
I think this is old, as the revenue var already has a proper nominal this should go ok.

note the comment is old, this is not hydrogen production, changed that